src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -393,6 +394,203 @@ internal string WorkingDirectory
             get { return _workingDirectory; }
         }
 
+        #endregion Internal properties
+
+        #region static methods
+        /// <summary>
+        /// Processes the -SettingFile Argument.
+        /// </summary>
+        /// <param name="args">
+        /// The command line parameters to be processed.
+        /// </param>
+        /// <param name="argIndex">
+        /// The index in args to the argument following '-SettingFile'.
+        /// </param>
+        /// <param name="parser">
+        /// Used to allow the helper to write errors to the console.  If not supplied, no errors will be written.
+        /// </param>
+        /// <returns>
+        /// Returns true if the argument was parsed successfully and false if not.
+        /// </returns>
+        private static bool TryParseSettingFileHelper(string[] args, int argIndex, CommandLineParameterParser parser)
+        {
+            if (argIndex >= args.Length)
+            {
+                if (parser != null)
+                {
+                    parser.WriteCommandLineError(
+                        CommandLineParameterParserStrings.MissingSettingsFileArgument);
+                }
+
+                return false;
+            }
+
+            string configFile = null;
+            try
+            {
+                configFile = NormalizeFilePath(args[argIndex]);
+            }
+            catch (Exception ex)
```

Comment:
I believe we could remove the `try-catch`. See my comment for NormalizeFilePath() method.